foam_controlwrapper/foam_mesh.py

```diff
@@ -657,9 +837,104 @@ def generate_uuidmapping(self, nPoints, nEdges, nFaces, nCells):
+        fvSolutionDict = fileContent['system/fvSolution']
+        controlDict = fileContent['system/controlDict'] + 'timePrecision 6;'
+
+        foamface.modifyNumerics(self.name, fvSchemesDict, fvSolutionDict, controlDict)
+
+
+    def modifyFields(self, BC):
+        """ Modifies the internal fields and boundary conditions
+        """
+        dimensionset = [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
+        nCells = foamface.getCellCount(self.name)
+        p_values = [0.0 for item in range(nCells)]
+        U_values = [(0.0, 0.0, 0.0) for item in range(nCells)]                                                
+
+        for cell in self.iter_cells():
+            p_values[self._uuidToFoamLabel[cell.uid]] = cell.data[CUBA.PRESSURE]
+            U_values[self._uuidToFoamLabel[cell.uid]] = cell.data[CUBA.VELOCITY]
+
+        foamface.setAllCellData(self.name, "p", dimensionset, p_values)
+        foamface.setAllCellVectorData(self.name, "U", dimensionset, U_values)
+
+        #Refresh boundary conditions
+        velocityBCs = BC[CUBA.VELOCITY]
+        myDict = ""
+        for boundary in velocityBCs:
+            myDict = myDict + str(boundary) + "\n{\n"               
+            if velocityBCs[boundary] == "zeroGradient":
+                myDict = myDict + "\t type \t zeroGradient;\n"
+                myDict = myDict + "\t value \t uniform (0 0 0);\n"
+            elif velocityBCs[boundary] == "empty":
+                myDict = myDict + "\t type \t empty;\n"
+            else:
+                myDict = myDict + "\t type \t fixedValue;\n"
+                myDict = myDict + "\t value \t uniform (" \
+                              + str(velocityBCs[boundary][0]) + " " \
+                              + str(velocityBCs[boundary][1]) + " " \
+                              + str(velocityBCs[boundary][2]) + ");\n"
+            myDict = myDict + "}\n"
+
+        foamface.setBC(self.name, "U", myDict)
+
+        pressureBCs = BC[CUBA.PRESSURE]
+        myDict = ""
+        for boundary in pressureBCs:
+            myDict = myDict + str(boundary) + "\n{\n"
+            if pressureBCs[boundary] == "zeroGradient":
+                myDict = myDict + "\t type \t zeroGradient;\n"
+                myDict = myDict + "\t value \t uniform 0;\n"
+            elif pressureBCs[boundary] == "fixedFluxPressure":
+                myDict = myDict + "\t type \t fixedFluxPressure;\n"
+                myDict = myDict + "\t value \t uniform 0;"
+            elif pressureBCs[boundary] == "empty":
+                myDict = myDict + "\t type \t empty;\n"
+            else:
+                myDict = myDict + "\t type \t fixedValue;\n"
+                myDict = myDict + "\t value \t uniform " + str(pressureBCs[boundary]) + ";\n"
+            myDict = myDict + "}\n"
+        
+        foamface.setBC(self.name, "p", myDict)
+
+
+    def run(self,nproc):
+        """ Run the required case
+        """
+        foamface.run(self.name,nproc)
+
+
+    def getXYZUVW(self):
```

Comment:
I think this method should be define somewhere else (i.e. outside of the FoamMesh).  Unless I am missing something, it looks like it can be rewritten to just use the methods of ABCMesh in order to "get" this XZUVW format.